### PR TITLE
feat: lens hidden comments

### PR DESCRIPTION
**Pull request by LeifXu**
# Pull Request

## Related Issue

Closes FW-765

## Checklist

-   [ ] I have tested these changes locally.
-   [ ] I have reviewed the code changes.
-   [ ] I have added unit tests if applicable.

### CODE REVIEW

File: src/components/Comments/index.tsx [hunk 1]
```diff
@@ -42,19 +44,31 @@ export const CommentList = memo<CommentListProps>(function CommentList({ postId,
...
+                    components: {
+                        Footer: CommentsFooter as any,
+                    },
+                }}
+                NoResultsFallbackProps={{
+                    icon: <MessageIcon width={24} height={24} />,
+                    message: <Trans>Be the first one to comment!</Trans>,
+                }}
+            />
+            {queryResult.data.length <= 0 && source === Source.Lens ? (
```
Comment (swkatmask) [inline src/components/Comments/index.tsx:69]: Might be not all comments hidden, but just some of them.

File: src/components/VirtualList/VirtualListFooterBottomText.tsx [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import { Trans } from '@lingui/macro';
+
+export function VirtualListFooterBottomText() {
+    return (
+        <div className="flex items-center justify-center p-6 text-base text-secondary">
+            <Trans>You&apos;ve hit rock bottom.</Trans>
```
Comment (swkatmask) [inline src/components/VirtualList/VirtualListFooterBottomText.tsx:6]:
```suggestion
            <Trans>You&apos;ve hit rock bottom.</Trans>
```

html entities do not work in jsx, you can replace with the character instead.

**Review by swkatmask — CHANGES_REQUESTED** [review]

File: src/components/ShowMoreComments.tsx [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+                            height={24}
+                            key={url}
+                            className="h-6 w-6 rounded-full"
+                            src={url}
+                            alt="avatar"
+                        />
+                    ))}
+                </div>
+            ) : null}
+            <div className="mx-3 text-[15px] font-bold leading-6">Show more comments</div>
```
Comment (guanbinrui) [inline src/components/ShowMoreComments.tsx:38]:
```suggestion
            <div className="mx-3 text-[15px] font-bold leading-6"><Trans>Show more comments</Trans></div>
```

File: src/components/ShowMoreComments.tsx [hunk 4]
```diff
@@ -0,0 +1,46 @@
+import LineArrowUpSVG from '@/assets/line-arrow-up.svg';
+import { Image } from '@/components/Image.js';
+import { classNames } from '@/helpers/classNames.js';
+
+export interface ShowMoreCommentsProps {
+    avatarUrls?: string[];
+    maxAvatarCount?: number | false;
```
Comment (guanbinrui) [inline src/components/ShowMoreComments.tsx:7]:
```suggestion
    maxAvatarCount?: number;
```

File: src/components/Comments/index.tsx [hunk 5]
```diff
@@ -42,19 +44,31 @@ export const CommentList = memo<CommentListProps>(function CommentList({ postId,
...
+                    listKey: `${ScrollListKey.Comment}:${postId}`,
+                    computeItemKey: (index, post) => `${post.postId}-${index}`,
+                    itemContent: (index, post) =>
+                        getPostItemContent(index, post, `${ScrollListKey.Comment}:${postId}`, { isComment: true }),
+                    context: {
+                        postId,
+                        source,
+                    },
+                    components: {
+                        Footer: CommentsFooter as any,
```
Comment (guanbinrui) [inline src/components/Comments/index.tsx:61]: Avoid `any`.

File: src/components/ShowMoreComments.tsx [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+    isOpen?: boolean;
+    onClick?: () => void;
+    className?: string;
```
Comment (guanbinrui) [inline src/components/ShowMoreComments.tsx:10]:
```suggestion
export interface ShowMoreCommentsProps extends HTMLProps<HTMLDivElement> {
    avatarUrls?: string[];
    maxAvatarCount?: number | false;
    isOpen?: boolean;
```

File: src/components/VirtualList/VirtualListFooterBottomText.tsx [hunk 7]
```diff
@@ -0,0 +1,9 @@
+import { t, Trans } from '@lingui/macro';
+
+export function VirtualListFooterBottomText() {
+    return (
+        <div className="flex items-center justify-center p-6 text-base text-secondary">
+            <Trans>{t`You've hit rock bottom.`}</Trans>
```
Comment (swkatmask) [inline src/components/VirtualList/VirtualListFooterBottomText.tsx:6]:
```suggestion
        {t`You've hit rock bottom.`}
```

File: src/components/Comments/index.tsx (hunk 1 above)
Comment (LeifXu) [reply src/components/Comments/index.tsx:69]: This is actually for the situation where there are no official comments, but there are hidden comments
IMAGE 2024-05-28 16:58:35

**Review by swkatmask — APPROVED** [review]

**Review by swkatmask — APPROVED** [review]

**Review by guanbinrui — APPROVED** [review]

**Pull request merged**